app/src/test/java/ch/hikemate/app/model/facilities/FacilitiesViewModelTest.kt

```diff
@@ -0,0 +1,183 @@
+        onFailure = { fail("Should not be called") })
+  }
+
+  @Test
+  fun testGetFacilities_onFailure() {
+    val exception = Exception("Test exception")
+
+    `when`(mockFacilitiesRepository.getFacilities(any(), any(), any())).then {
+      val onFailure = it.getArgument<(Exception) -> Unit>(2)
+      onFailure(exception)
+    }
+
+    facilitiesViewModel.getFacilities(
+        bounds = testBounds,
+        onSuccess = { fail("Should not be called") },
+        onFailure = { ex -> assertEquals(exception, ex) })
+  }
+
+  @Test
+  fun testGetFacilities_usesCache_sameBounds() = runTest {
+    val facilities = listOf(Facility(FacilityType.TOILETS, GeoPoint(46.51, 6.61)))
+
+    var onSuccessCallCount = 0 // Used to make sure the first onSuccess callback is only called once
+
+    `when`(mockFacilitiesRepository.getFacilities(any(), any(), any())).then {
+      val onSuccess = it.getArgument<(List<Facility>) -> Unit>(1)
+      onSuccess(facilities)
+    }
+
+    facilitiesViewModel.getFacilities(
+        bounds = testBounds,
+        onSuccess = { result ->
+          assertEquals(facilities, result)
+          onSuccessCallCount++
+        },
+        onFailure = { fail("Should not be called") })
+
+    testDispatcher.scheduler.advanceUntilIdle()
+
+    // Second call to getFacilities should use the cache
+    facilitiesViewModel.getFacilities(
+        bounds = testBounds,
+        onSuccess = { result -> assertEquals(facilities, result) },
+        onFailure = { fail("Should not be called") })
+
+    testDispatcher.scheduler.advanceUntilIdle()
+
+    verify(mockFacilitiesRepository, times(1)).getFacilities(any(), any(), any())
+    assertEquals(1, onSuccessCallCount)
+  }
+
+  @Test
+  fun testGetFacilities_usesCache_containedBounds() = runTest {
+    val facilities = listOf(Facility(FacilityType.TOILETS, GeoPoint(46.505, 6.605)))
+
+    var onSuccessCallCount = 0 // Used to make sure the first onSuccess callback is only called once
+
+    `when`(mockFacilitiesRepository.getFacilities(any(), any(), any())).then {
+      val onSuccess = it.getArgument<(List<Facility>) -> Unit>(1)
+      onSuccess(facilities)
+    }
+
+    facilitiesViewModel.getFacilities(
+        bounds = testBounds,
+        onSuccess = { result ->
+          assertEquals(facilities, result)
+          onSuccessCallCount++
+        },
+        onFailure = { fail("Should not be called") })
+
+    testDispatcher.scheduler.advanceUntilIdle()
+
+    // Second call to getFacilities should use the cache. The bounds here are within the previous
+    // calls bounds
+    facilitiesViewModel.getFacilities(
+        bounds = Bounds(46.501, 6.601, 46.509, 6.619),
+        onSuccess = { result -> assertEquals(facilities, result) },
+        onFailure = { fail("Should not be called") })
+
+    testDispatcher.scheduler.advanceUntilIdle()
+
+    verify(mockFacilitiesRepository, times(1)).getFacilities(any(), any(), any())
+    assertEquals(1, onSuccessCallCount)
```

Comment:
Warning: Same remark as above. This should be 2

Reply:
done